tests/test_mpi_settings.py

```diff
@@ -0,0 +1,272 @@
+import logging
+import os
+import os.path as osp
+import stat
+import sys
+from asyncio.log import logger
+from unicodedata import name
+
+import pytest
+
+from smartsim.error import SSUnsupportedError, LauncherError
+from smartsim.settings.mpiSettings import (
+    MpiexecSettings,
+    MpirunSettings,
+    OrterunSettings,
+    _BaseMPISettings,
+)
+
+
+@pytest.mark.parametrize(
+    "MPISettings", [MpirunSettings, MpiexecSettings, OrterunSettings]
+)
+def test_not_instanced_if_not_found(MPISettings):
+    old_path = os.getenv("PATH")
+    try:
+        os.environ["PATH"] = ""
+        with pytest.raises(LauncherError):
+            MPISettings(sys.executable)
+    finally:
+        os.environ["PATH"] = old_path
+
+
+@pytest.mark.parametrize(
+    "MPISettings,stubs_path,stub_exe",
+    [
+        pytest.param(
+            MpirunSettings,
+            osp.join("mpi_impl_stubs", "openmpi4"),
+            "mpirun",
+            id="OpenMPI4-mpirun",
+        ),
+        pytest.param(
+            MpiexecSettings,
+            osp.join("mpi_impl_stubs", "openmpi4"),
+            "mpiexec",
+            id="OpenMPI4-mpiexec",
+        ),
+        pytest.param(
+            OrterunSettings,
+            osp.join("mpi_impl_stubs", "openmpi4"),
```

Comment:
Should we add parameters to this test to make sure the stubs of intel's of mpiexec/mpirun/orterun do not throw a warning? If not I think we can remove those stubs

Reply:
I suppose we could, but at this point I sort of see this as the complementary test of the slurm.mpiexec one. I'm more inclined to do as you suggest and remove the Intel ones since we probably only need to test one MPI implementation.